packages/template-retail-react-app/app/pages/account/wishlist/index.jsx

```diff
@@ -21,69 +21,93 @@ import WishlistPrimaryAction from './partials/wishlist-primary-action'
 import WishlistSecondaryButtonGroup from './partials/wishlist-secondary-button-group'
 
 import {API_ERROR_MESSAGE} from '../../../constants'
+import {useCurrentCustomer} from '../../../hooks/use-current-customer'
 
 const numberOfSkeletonItems = 3
 
 const AccountWishlist = () => {
-    const customer = useCustomer()
     const navigate = useNavigation()
     const {formatMessage} = useIntl()
     const toast = useToast()
+
     const [selectedItem, setSelectedItem] = useState(undefined)
-    const [localQuantity, setLocalQuantity] = useState({})
     const [isWishlistItemLoading, setWishlistItemLoading] = useState(false)
-    const wishlist = useWishlist()
+
+    const {data: wishListData, isLoading: isWishListLoading} = useWishList()
+    const productIds = wishListData?.customerProductListItems?.map((item) => item.productId)
+
+    const {data: productsData, isLoading: isProductsLoading} = useProducts(
+        {parameters: {ids: productIds?.join(','), allImages: true}},
+        {enabled: productIds?.length > 0}
+    )
+
+    const wishListItems = wishListData?.customerProductListItems?.map((item, i) => {
+        return {
+            ...item,
+            product: productsData?.data?.[i]
+        }
+    })
+
+    const updateCustomerProductListItem = useShopperCustomersMutation(
+        'updateCustomerProductListItem'
+    )
+    const deleteCustomerProductListItem = useShopperCustomersMutation(
+        'deleteCustomerProductListItem'
+    )
+    const {data: customer} = useCurrentCustomer()
 
     const handleActionClicked = (itemId) => {
-        setWishlistItemLoading(!!itemId)
+        setWishlistItemLoading(Boolean(itemId))
         setSelectedItem(itemId)
     }
 
     const handleItemQuantityChanged = async (quantity, item) => {
-        // This local state allows the dropdown to show the desired quantity
-        // while the API call to update it is happening.
-        setLocalQuantity({...localQuantity, [item.productId]: quantity})
-        setWishlistItemLoading(true)
+        let isValidChange = false
         setSelectedItem(item.productId)
+
+        const body = {
+            ...item,
+            quantity: parseInt(quantity)
+        }
+        // To meet expected schema, remove the custom `product` we added
+        delete body.product
+
+        const parameters = {
+            customerId: customer.customerId,
+            itemId: item.id,
+            listId: wishListData?.id
+        }
+
+        const mutation =
+            parseInt(quantity) > 0
+                ? updateCustomerProductListItem.mutateAsync({body, parameters})
+                : deleteCustomerProductListItem.mutateAsync({parameters})
```

Comment:
What happens if you do `update` with 0? Feels like the back end should be capable of handling the deletion logic when quantity is 0.